Code/src/main/java/nl/utwente/viskell/ui/components/Block.java

```diff
@@ -116,51 +118,65 @@ public final CustomUIPane getPane() {
     }
 
     /**
+     * @return All InputAnchors of the block.
+     */
+    public List<InputAnchor> getAllInputs() {
+        return ImmutableList.of();
+    }
+
+    /**
+     * @return Only the active (as specified with by the knot index) inputs.
+     */
+    public List<InputAnchor> getActiveInputs() {
+        return getAllInputs();
+    }
+    
+    /**
+     * @return the optional output Anchor for this Block
+     */
+    public Optional<OutputAnchor> getOutputAnchor() {
```

Comment:
Could consider making this

    public List<OutputAnchor> getOutputAnchors()

if we want to go N inputs/N outputs in the future.